### ISSUE TITLE
PyTorch BERT gradient clipping

### ISSUE THREAD

**Issue by eric-haibin-lin** [NVIDIA/DeepLearningExamples#287]
I noticed that pytorch bert pre-training clips the gradient on a per layer basis: 
https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/optimization.py#L359-L360 

meanwhile the tensorflow implementation clips the global gradient norm: https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow/LanguageModeling/BERT/optimization.py#L131-L136 

Is there a reason why clipping per layer is used in PyTorch? Is there any convergence issue with global norm clipping in PyTorch? Or because clip per layer leads to better convergence than global norm clipping? Thanks

**Comment by sharathts** [comment]
The pretraining script uses the LAMB optimizer and it clips the global gradient norm https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/optimization.py#L148

On the other hand, implementation in BertAdam is incorrect and should be fixed to clip globally. I will update it soon. Thank you for pointing it out.

**Comment by eric-haibin-lin** [comment]
Thanks! @sharathts

**Comment by sharathts** [comment]
Fixed here: https://github.com/NVIDIA/DeepLearningExamples/pull/293
